pkgs/applications/audio/uade123/default.nix

```diff
@@ -1,25 +1,74 @@
-{ lib, stdenv, fetchurl, which, libao, pkg-config }:
+{ lib
+, stdenv
+, fetchFromGitLab
+, pkg-config
+, which
+, makeWrapper
+, libao
+, libbencodetools
+, sox
+, lame
+, flac
+, vorbis-tools
+}:
 
 let
-  version = "2.13";
-in stdenv.mkDerivation {
+  binPath = lib.makeBinPath [ sox lame flac vorbis-tools ];
+in
+stdenv.mkDerivation {
   pname = "uade123";
-  inherit version;
-  src = fetchurl {
-    url = "http://zakalwe.fi/uade/uade2/uade-${version}.tar.bz2";
-    sha256 = "04nn5li7xy4g5ysyjjngmv5d3ibxppkbb86m10vrvadzxdd4w69v";
+  version = "unstable-2021-05-21";
+
+  src = fetchFromGitLab {
+    owner = "uade-music-player";
+    repo = "uade";
+    rev = "7169a46e777d19957cd7ff8ac31843203e725ddc";
+    sha256 = "1dm7c924fy79y3wkb0qi71m1k6yw1x6j3whw7d0w4ka9hv6za03b";
   };
-  nativeBuildInputs = [ pkg-config which ];
-  buildInputs = [ libao ];
+
+  postPatch = ''
+    patchShebangs .
+    substituteInPlace src/frontends/mod2ogg/mod2ogg2.sh.in \
+      --replace '-e stat' '-n stat'
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+    which
+    makeWrapper
+  ];
+
+  buildInputs = [
+    libao
+    libbencodetools
+    sox
+    lame
+    flac
+    vorbis-tools
+  ];
+
+  configureFlags = [
+    "--bencode-tools-prefix=${libbencodetools}"
+  ];
 
   enableParallelBuilding = true;
+
   hardeningDisable = [ "format" ];
 
+  postInstall = ''
+    wrapProgram $out/bin/mod2ogg2.sh \
```

Comment:
Scripts under bin should not have an extension.

Reply:
That script has been part of the installation step of the project with its .sh extension intact for almost 16 years, I don't see the point in deviating from that now. Changing it would likely be an annoyance more than anything else IMO.

Reply:
Than it is wrong for 16 years. Binaries should not have an extensions like py, rb, pl or sh.

> Changing it would likely be an annoyance more than anything else IMO.

Nope, it wouldn't because binaries are expected to not have a file extension.

Alsa-info is shipped by Debian/Ubuntu like this: ``/usr/sbin/alsa-info`` and should also be changed to alsa-info in nixos. Thanks for the finding.

Reply:
I disagree. There is no rule that executables in bin must have no extension. In Nixpkgs alone there are thousands of scripts under bin that retain their extension. And no other big distro / package manager with an UADE package removes the extension from this script. Why would you want to introduce inconsistency?

I want to maintain this package, and I think the extension is fine. :)

Reply:
Thats wrong. Debian removes the script extensions but whatever, I will patch the things that I use over time.

https://www.debian.org/doc/debian-policy/ch-files.html#scripts

> When scripts are installed into a directory in the system PATH, the script name should not include an extension such as .sh or .pl that denotes the scripting language currently used to implement it.